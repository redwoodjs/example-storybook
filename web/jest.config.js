// More info at https://redwoodjs.com/docs/project-configuration-dev-test-build

const config = {
  rootDir: '../',
  preset: '@redwoodjs/testing/config/jest/web',
  verbose: true,
  moduleNameMapper: {
    // mock redwood magic webpack aliases
    // see: https://github.com/redwoodjs/redwood/blob/main/packages/testing/config/storybook/preview.js
    '~__REDWOOD__USER_WEB_DEFAULT_CSS': 'identity-obj-proxy',
    '~__REDWOOD__USER_STORYBOOK_PREVIEW_CONFIG':
      '<rootDir>/web/config/storybook.preview.js',
  },
  transform: {
    '^.+\\.mdx?$': '@storybook/addon-docs/jest-transform-mdx', // https://github.com/storybookjs/storybook/issues/11376#issuecomment-757417250
  },
  setupFilesAfterEnv: [`<rootDir>/web/test/mocks.js`],
}

module.exports = config
