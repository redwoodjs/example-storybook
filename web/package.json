{
  "name": "web",
  "version": "0.0.0",
  "private": true,
  "browserslist": {
    "development": [
      "last 1 version"
    ],
    "production": [
      "defaults",
      "not IE 11",
      "not IE_Mob 11"
    ]
  },
  "dependencies": {
    "@redwoodjs/forms": "0.49.0",
    "@redwoodjs/router": "0.49.0",
    "@redwoodjs/web": "0.49.0",
    "prop-types": "15.8.1",
    "react": "17.0.2",
    "react-dom": "17.0.2"
  }
}
