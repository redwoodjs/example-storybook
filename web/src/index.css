/**
 * START --- SETUP TAILWINDCSS EDIT
 *
 * `yarn rw setup ui tailwindcss` placed these imports here
 * to inject Tailwind's styles into your CSS.
 * For more information, see: https://tailwindcss.com/docs/installation#include-tailwind-in-your-css
 */
@import 'tailwindcss/base';
@import 'tailwindcss/components';
@import 'tailwindcss/utilities';
/**
 * END --- SETUP TAILWINDCSS EDIT
 */
body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
}
ul {
  list-style-type: none;
  margin: 1rem 0;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 1rem 0 0 ;
}
h1 > a {
  text-decoration: none;
  color: black;
}
button, input, label, textarea {
  display: block;
  outline: none;
}
label {
  margin-top: 1rem;
}
.error {
  color: red;
}
input.error, textarea.error {
  border: 1px solid red;
}
.form-error {
  color: red;
  background-color: lavenderblush;
  padding: 1rem;
  display: inline-block;
}
.form-error ul {
  list-style-type: disc;
  margin: 1rem;
  padding: 1rem;
}
.form-error li {
  display: list-item;
}
.flex-between {
  display: flex;
  justify-content: space-between;
}
.flex-between button {
  display: inline;
}
