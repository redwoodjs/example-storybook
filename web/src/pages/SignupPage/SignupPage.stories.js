//@ts-check
import SignupPage from './SignupPage'

export default {
  title: 'Pages/SignupPage',
  component: SignupPage,
}
const Template = (args) => <SignupPage {...args} />

export const generated = Template.bind({})
