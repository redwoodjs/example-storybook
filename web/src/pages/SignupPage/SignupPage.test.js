import { render } from '@redwoodjs/testing/web'

import SignupPage from './SignupPage'

//   Improve this test with help from the Redwood Testing Doc:
//   https://redwoodjs.com/docs/testing#testing-pages-layouts

describe('SignupPage', () => {
  it('renders successfully', () => {
    expect(() => {
      render(<SignupPage />)
    }).not.toThrow()
  })

  it('fragment matches snapshot', () => {
    const { asFragment } = render(<SignupPage />)
    expect(asFragment()).toMatchSnapshot()
  })
})
