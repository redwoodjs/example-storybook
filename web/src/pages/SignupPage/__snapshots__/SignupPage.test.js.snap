// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupPage fragment matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="rw-main w-96 mx-auto mt-12"
  >
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
    <div
      class="rw-scaffold rw-login-container"
    >
      <div
        class="rw-segment"
      >
        <header
          class="rw-segment-header"
        >
          <h2
            class="rw-heading rw-heading-secondary"
          >
            Signup
          </h2>
        </header>
        <div
          class="rw-segment-main"
        >
          <div
            class="rw-form-wrapper"
          >
            <form
              class="rw-form-wrapper"
            >
              <label
                class="rw-label"
                for="username"
              >
                Username
              </label>
              <input
                class="rw-input"
                id="username"
                name="username"
                type="text"
              />
              <label
                class="rw-label"
                for="password"
              >
                Password
              </label>
              <input
                autocomplete="current-password"
                class="rw-input"
                id="password"
                name="password"
                type="password"
              />
              <div
                class="rw-button-group"
              >
                <button
                  class="rw-button rw-button-blue"
                  type="submit"
                >
                  Sign Up
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        class="rw-login-link mt-2 text-center"
      >
        <span>
          Already have an account?
        </span>
         
        <a
          class="rw-link"
          href="/login"
        >
          Log in!
        </a>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
